src/views/Main/Catalog/Components/NameNamespaceAutosuggest.js

```diff
@@ -0,0 +1,161 @@
+import React, { PropTypes } from 'react';
+import Autosuggest from 'react-autosuggest';
+import { Button } from 'react-bootstrap';
+import '../../../../styles/autosuggest.css';
+
+const getSuggestionValue = suggestion => suggestion;
+
+const renderSuggestion = suggestion => (
+  <div>
+    {suggestion}
+  </div>
+);
+
+export class NameNamespaceAutosuggest extends React.Component {
+  static propTypes = {
+    id: PropTypes.number,
+    names: PropTypes.object,
+    namespaces: PropTypes.object,
+    className: PropTypes.string,
+    addProperty: PropTypes.func,
+    type: PropTypes.string
+  }
+
+  constructor() {
+    super();
+    this.state = {
+      nameVal: '',
+      nameSuggestions: [],
+      namespaceVal: '',
+      namespaceSuggestions: []
+    };
+  }
+
+  getNamespaceVal = () => document
+    .getElementById('newNamespace'
+      .concat(this.props.type)
+      .concat(this.props.id)
+    ).firstChild
```

Comment:
The issue was that the input element was being created by react-autosuggest and was a couple layers deep, but I just realized I should just be using the state for this anyway. Oops. Thanks